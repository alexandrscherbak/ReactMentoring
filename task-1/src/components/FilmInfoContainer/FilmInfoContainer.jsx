import React from 'react';
import FilmGenre from './FilmGenre/FilmGenre.jsx';
import FilmImage from './FilmImage/FilmImage.jsx';
import FilmReleaseDate from './FilmReleaseDate/FilmReleaseDate.jsx';
import FilmTitle from './FilmTitle/FilmTitle.jsx';

const FilmInfoContainer = ({filmInfo, selectMovie}) => (
	<div className="film-info-container" onClick={() => selectMovie(filmInfo)}>
		<FilmImage imageUrl={filmInfo.poster_path} title={filmInfo.title}></FilmImage>
		<div className="title-and-date">
			<FilmTitle title={filmInfo.title}></FilmTitle>
			<FilmReleaseDate releaseDate={filmInfo.release_date}></FilmReleaseDate>
		</div>
		<FilmGenre genre={filmInfo.genres[0]}></FilmGenre>

		<style jsx>
			{`
				.film-info-container {
					width: 300px;
					height: 600px;
					cursor: pointer;
				}
				.title-and-date {
					display: flex;
					justify-content: space-between;
				}
			`}
		</style>
	</div>
);

export default FilmInfoContainer;
