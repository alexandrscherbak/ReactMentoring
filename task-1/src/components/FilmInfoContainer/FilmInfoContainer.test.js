import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmInfoContainer from './FilmInfoContainer.jsx';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const props = {
	filmInfo: {
		poster_path: 'poster_path',
		title: 'film',
		release_date: '2018-04-01',
		genres: ['action', 'drama'],
	},
	selectMovie: jest.fn(),
};

function setup() {
	return mount(<FilmInfoContainer {...props}/>);
}

describe('FilmInfoContainer', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<FilmInfoContainer {...props}/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should call selectMovie if button clicked', () => {
		const enzymeWrapper = setup();
		const filmInfoContainer = enzymeWrapper.find('.film-info-container');
		expect(props.selectMovie.mock.calls.length).toBe(0);
		filmInfoContainer.props().onClick();
		expect(props.selectMovie.mock.calls.length).toBe(1);
	});
});
