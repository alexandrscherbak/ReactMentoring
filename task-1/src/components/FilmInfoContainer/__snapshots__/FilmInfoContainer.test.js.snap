// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmInfoContainer snapshot 1`] = `
<div
  className="jsx-2546762946 film-info-container"
  onClick={[Function]}
>
  <img
    alt="film"
    className="jsx-2994430438"
    src="poster_path"
  />
  <div
    className="jsx-2546762946 title-and-date"
  >
    <div
      className="jsx-433743616"
    >
      <span
        className="jsx-433743616"
      >
        film
      </span>
    </div>
    <div
      className="jsx-617062702"
    >
      <span
        className="jsx-617062702"
      >
        2018
      </span>
    </div>
  </div>
  <div
    className="jsx-672986837"
  >
    <span
      className="jsx-672986837"
    >
      action
    </span>
  </div>
</div>
`;
